uint8 port_no
bool is_on